## Importing the libraries

In [1]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

## Data Preprocessing

In [2]:
dataset = pd.read_csv('Market_Basket_Optimisation.csv', header = None)


In [3]:
transactions = []
for i in range(0, 7501):
    transactions.append([str(dataset.values[i,j]) for j in range(0, 20)])

## Training Apriori on the dataset

In [4]:
from apyori import apriori

In [5]:
rules = apriori(transactions, min_support = 0.003, min_confidence = 0.2, min_lift = 3, min_length = 2)

## Results

In [6]:
results = list(rules)

In [7]:
print(results[0])

RelationRecord(items=frozenset({'light cream', 'chicken'}), support=0.004532728969470737, ordered_statistics=[OrderedStatistic(items_base=frozenset({'light cream'}), items_add=frozenset({'chicken'}), confidence=0.29059829059829057, lift=4.84395061728395)])


## Conclusion

First Association Rule Explanation: 
the light cream and chicken are bought frequently. the support is 0.0045(calculated by dividing the no. of transactions containing light cream divided by total no. of transacitons) the confidence for this rule is 0.29 which means that out of all transactions containing light cream, 29% of the transcaitons are likely to contain chicken as well. The lift of 4.84 shows that the chicken is 4.84 more likely to be bought by the customers that buy light cream compared to its individual sale.